plugins/docker/user_login.go

```diff
@@ -0,0 +1,85 @@
+package docker
+
+import (
+	"context"
+	"encoding/base64"
+	"strings"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func UserLogin() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.UserLogin,
+		DocsURL:       sdk.URL("https://docker.com/docs/user_login"),              // TODO: Replace with actual URL
+		ManagementURL: sdk.URL("https://console.docker.com/user/security/tokens"), // TODO: Replace with actual URL
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Username of the user registered in Docker.",
+			},
+			{
+				Name:                fieldname.Password,
+				MarkdownDescription: "Password of the user registered in Docker.",
+				Secret:              true,
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+
+			TryDockerConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
```

Comment:
I'm not able to find anywhere documentation related to this authentication method for docker against DockerHub (or other platforms).

Can you please provide some documentation on this?

Reply:
yess here it is https://docs.docker.com/engine/reference/commandline/login/